### PR TITLE
Use methods from KissXML instead of re-implemting them

### PR DESCRIPTION
**Pull request by t-unit**
Integrating XMPPFramework using Swift PM (using a fork of this framework) does expose a lot of warnings about duplicate implementations:

```
ld: warning: method ‘-elementForName:’ in category from [...]/KissXML.o conflicts with same method from another category
ld: warning: method ‘-elementForName:xmlns:’ in category from [...]/KissXML.o conflicts with same method from another category
ld: warning: method ‘-xmlns’ in category from [...]/KissXML.o conflicts with same method from another category
ld: warning: method ‘-setXmlns:’ in category from [...]/KissXML.o conflicts with same method from another category
ld: warning: method ‘-prettyXMLString’ in category from [...]/KissXML.o conflicts with same method from another category
ld: warning: method ‘-compactXMLString’ in category from [...]/KissXML.o conflicts with same method from another category
ld: warning: method ‘-addAttributeWithName:stringValue:’ in category from [...]/KissXML.o conflicts with same method from another category
ld: warning: method ‘-attributesAsDictionary’ in category from [...]/KissXML.o conflicts with same method from another category
ld: warning: method ‘+elementWithName:xmlns:’ in category from [...]/KissXML.o conflicts with same method from another category
```

Changes in this pull request remove the duplicate implementations and directly use KissXML instead if running on iOS.

### CODE REVIEW

File: Categories/NSXMLElement+XMPP.m
Comment (chrisballinger) [inline Categories/NSXMLElement+XMPP.m:179]: It doesn't look like anything was changed here other than whitespace. If possible please revert the whitespace changes to minimize the diff

**Review by chrisballinger — CHANGES_REQUESTED** [review]
@t-unit Before merging, there are a few changes that would be great if you have the time:

* Fix merge conflicts
* Please use framework imports like `#import <KissXML/KissXML.h>` instead of module imports in the XMPPFramework library code. Module imports are fine in the test targets.
* Please minimize the diff as much as possible by reverting whitespace changes, or code reorganization

Thanks!

**Comment by t-unit** [comment]
@chrisballinger pls have another look. Most whitespace changes are removed. (What is left is mostly trailing whitespace.)

**Comment by chrisballinger** [comment]
Thanks! Will take another look when the tests are passing again

**Comment by t-unit** [comment]
@chrisballinger Fixed the issues introduced by my changes. Also tried to fix the test on Travis, but now it seems to fails with some unrelated errors...

**Comment by chrisballinger** [comment]
@t-unit Thanks! Looks like it's getting caught up here on the macOS / Carthage / Swift tests. Since Carthage isn't seeing much active maintenance anymore, I think it might be worth commenting out the Carthage tests for now if the fix isn't quick and straightforward.

Once it's passing the remaining tests (it looks like everything else is passing), I think this will probably be good to merge.

```
xcodebuild -project Xcode/Testing-Carthage/XMPPFrameworkSwiftTests.xcodeproj -scheme "XMPPFrameworkTests (macOS)" -sdk macosx -arch x86_64 test | xcpretty -c
 Multiple targets match implicit dependency for product reference 'XMPPFramework.framework'. Consider adding an explicit dependency on the intended target to resolve this ambiguity. (in target 'XMPPFrameworkSwift (macOS)' from project 'XMPPFramework')
25079
25080❌  error: Multiple commands produce '/Users/travis/Library/Developer/Xcode/DerivedData/XMPPFrameworkSwiftTests-gjimoldmrjqhzvfvfwmkfzozjaqn/Build/Products/Debug/CocoaLumberjack.framework/Versions/A/CocoaLumberjack':
25081
25082
25083
25084❌  error: Multiple commands produce '/Users/travis/Library/Developer/Xcode/DerivedData/XMPPFrameworkSwiftTests-gjimoldmrjqhzvfvfwmkfzozjaqn/Build/Products/Debug/CocoaAsyncSocket.framework/Versions/A':
25085
25086
25087
25088❌  error: Multiple commands produce '/Users/travis/Library/Developer/Xcode/DerivedData/XMPPFrameworkSwiftTests-gjimoldmrjqhzvfvfwmkfzozjaqn/Build/Products/Debug/libidn.framework/Versions/A/libidn':
```

**Comment by t-unit** [comment]
@chrisballinger test now pass after commenting out the carthage tests

File: Extensions/Roster/MemoryStorage/XMPPUserMemoryStorageObject.m
Comment (chrisballinger) [inline Extensions/Roster/MemoryStorage/XMPPUserMemoryStorageObject.m:50]: Is this change necessary?

**Review by chrisballinger — APPROVED** [review]
Thanks for moving this forward! Looks good to merge once that `mutableCopy` is clarified.

File: Extensions/Roster/MemoryStorage/XMPPUserMemoryStorageObject.m
Comment (Neustradamus) [reply Extensions/Roster/MemoryStorage/XMPPUserMemoryStorageObject.m:50]: @t-unit: Have you looked the @chrisballinger comment?

Comment (t-unit) [reply Extensions/Roster/MemoryStorage/XMPPUserMemoryStorageObject.m:50]: @chrisballinger KissXML does expose this method with an immutable return value – see https://github.com/robbiehanson/KissXML/blob/6db0bcbb5a15687846f0c1a0db8d0fa5a2d3a7dc/KissXML/Additions/DDXMLElementAdditions.h#L23

Comment (chrisballinger) [reply Extensions/Roster/MemoryStorage/XMPPUserMemoryStorageObject.m:50]: @t-unit Yes exactly, so wouldn't we want to revert the `mutableCopy` changes on this line?

Comment (t-unit) [reply Extensions/Roster/MemoryStorage/XMPPUserMemoryStorageObject.m:50]: @chrisballinger Don't quite understand. The [`itemAttributes`](https://github.com/robbiehanson/XMPPFramework/blob/c20cafccd422c513f78a108aa20fbce9b79ab7e8/Extensions/Roster/MemoryStorage/XMPPUserMemoryStorageObject.h#L17) property is defined a mutable – so it should not get an immutable dictionary assigned to it.

Comment (chrisballinger) [reply Extensions/Roster/MemoryStorage/XMPPUserMemoryStorageObject.m:50]: @t-unit Ahhhh now I get it, I wasn't looking closely enough. The reason for the change is that bundled internal version of `attributesAsDictionary` returns a mutable value, and the very similar KissXML version returns it as immutable. Looks good to me!

**Comment by iAmcR** [comment]
Hi, I'm still seeing the conflict issues (warnings) in the latest version when using Swift Package manager. What would be the best way to get rid of those warnings?

**Pull request merged**
